conan/tools/meson/meson.py

```diff
@@ -1,40 +1,36 @@
 import os
 
 from conan.tools.build import build_jobs
-from conan.tools.cross_building import cross_building
 from conan.tools.meson import MesonToolchain
 
 
 class Meson(object):
-    def __init__(self, conanfile, build_folder='build'):
+    def __init__(self, conanfile):
         self._conanfile = conanfile
-        self._build_folder = build_folder
 
-    @property
-    def _build_dir(self):
-        build = self._conanfile.build_folder
-        if self._build_folder:
-            build = os.path.join(self._conanfile.build_folder, self._build_folder)
-        return build
-
-    def configure(self, source_folder=None):
-        source = self._conanfile.source_folder
-        if source_folder:
-            source = os.path.join(self._conanfile.source_folder, source_folder)
+    def configure(self,  build_script_folder=None):
```

Comment:
I doubt in meson world they call it `build script folder`, might be misleading for the people

Reply:
Suggestion for an alternative name?

Reply:
in [documentation](https://mesonbuild.com/Commands.html#configure), they refer to it as a `builddir`